### PR TITLE
Make REGoth compilable with latest bgfx

### PR DESCRIPTION
**Pull request by Kukunin**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [submodule "lib/bgfx-cmake"]
 	path = lib/bgfx-cmake
-	url = https://github.com/degenerated1123/bgfx-cmake.git
+	url = https://github.com/Kukunin/bgfx-cmake.git
```
Comment (degenerated1123) [inline .gitmodules]: You shouldn't change the repository url, please keep it to what it was before

**Comment by Kukunin** [comment]
Currently we have 3 bugs with this: 
- UI is black, but fully working (you can press enter etc, it just doesn't render)
- Everything is grey-scaled
- There is no bodies
 
screen shot 2017-10-06 at 14 32 45

On other hand - it works under MacOS #241

File: .gitmodules (hunk 1 above)
Comment (Kukunin) [reply .gitmodules]: I know, I just changed temporarily in this PR, while you didn't merge this into upstream

**Comment by Kukunin** [comment]
After recompilation of shaders with latest bgfx, the latest result is

image

**Comment by Drako** [comment]
For whatever reason the PC is missing his right leg xD

File: src/content/Shader.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#include "Shader.h"
+#include <utils/Utils.h>
+
+bgfx::ShaderHandle Shader::loadShader(const char* basePath, const char* _name)
+{
+    char filePath[512];
```
Comment (Drako) [inline src/content/Shader.cpp]: why not std::string?

File: src/content/Shader.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+        case bgfx::RendererType::OpenGLES:
+            shaderPath = "shaders/essl/";
+            break;
+
+        default:
+            break;
+    }
+
+    strcpy(filePath, basePath);
```
Comment (Drako) [inline src/content/Shader.cpp]: why not strncpy and strncat?

Comment (degenerated1123) [reply src/content/Shader.cpp]: That is based on the bgfx sample framework. We should rewrite that :)

**Comment by degenerated1123** [comment]
Found the issue causing the missing legs! 

In vs_skinned.sc, put `#define BGFX_CONFIG_MAX_BONES 100`, before including `common.h`, so it's like this:

```cpp
 // ...

#define LOCAL_POSITIONS_1 a_texcoord1
#define LOCAL_POSITIONS_2 a_texcoord2
#define LOCAL_POSITIONS_3 a_texcoord3
#define LOCAL_POSITIONS_4 a_texcoord4
#define BONE_INDICES a_texcoord5
#define WEIGHTS a_texcoord6

#define BGFX_CONFIG_MAX_BONES 100

#include "../common/common.sh"
#include "tools.sh"

uniform vec4 u_color;

// ...
```

**Comment by Kukunin** [comment]
Pushed the fix. Hopefully, after CI is done we can merge it

**Comment by degenerated1123** [comment]
Works on my machine *(tm)*

**Pull request merged**